### ISSUE TITLE
Refine templating to use generic form and dictionary

### ISSUE THREAD

**Issue by jeffbaumes** [XDATA-Year-3/vcharts#22]
Instead of special keys like `"{{"`, `"[["`, etc., the proposal is to have a set of defined functions in an object, such as:

```js
{
    lookup: function(variable, default) {
        // ...
    },
    equals: function(a, b) {
        return a === b;
    },
    map: function(array, loopVariable, item) {
        // ...
    }
}
```

Then the template explicitly calls each by name with a distinguishing prefix or suffix:

```json
{"lookup()": ["option", 5]}

{"equals()": [5, 10]}

{"if()": [false, "true-condition", "false-condition"]}
```

**Comment by jeffbaumes** [comment]
We could also just go all the way to Lisp by making the function the first part of the list

```json
["lookup()", "option", 5]

["equals()", 5, 10]

["if()", false, "true-condition", "false-condition"]
```

**Comment by jeffbaumes** [comment]
Resolved by #26
